src/frontend/src/env/rest/coingecko.env.ts

```diff
@@ -0,0 +1,5 @@
+import { UrlSchema } from '$lib/validation/url.validation';
+
+export const COINGECKO_API_URL = UrlSchema.parse('https://pro-api.coingecko.com/api/v3/');
```

Comment:
`parse` can throw an error at runtime which can per extension lead to the all app to not mount since this is executed at the root of the module.

We should probably rather use `safeParse`?

Reply:
my idea is that since they are hard-coded, the error would appear safely with CIs failing, in any PR that changes/updates/modifies it

so we would be aware of it, before merging a possible not-valid-URL

with `safeParse`, if the URL is wrongly parsed, the app would work either way, but only the `COINGECKO_API` functions will possibly fail

Reply:
How the CI would fail? `safe` is a runtime feature.

Reply:
What I meant was that with `parse` will fail, and that would be good

### `parse`

- CI runs and find an error in parsing --> fails --> we know that the URL we wrote in the code is somehow wrong

### `safeParse`

```typescript
export const COINGECKO_API_URL = UrlSchema.safeParse(...).success ? ... : fallbackValue;
```

- CI runs --> there is an error in parsing, but it falls back on the fallback value --> we think it is all good, but the Coingecko features are not working


Or you mean to not have a fallback value but raise the error? Maybe I misunderstood what you meant as implementation

NOTE: with E2E tests obviously this is not a discussion eheh

Reply:
I guess the CI don't run E2E tests against the production build, hence the CI won't run the production application and it can potentially ends with an application that does not start / mount.

It would be an edge case but, it's still a possible case therefore I think it's not safe to actually use `parse` at the root of a module / of the application without catching potential errors.

Reply:
yes, makes sense! What do you think would be better then? Throw an error in case of parsing unsuccessful or just fallback to an empty string?

Reply:
What's the current behavior if someone forgot to setup an env variable, empty string?
Should we just keep that approach (for the best or worst) and iterate if necessary?

Reply:
it would default to an `undefined`, so I keep the same; have a look at last commit

Reply:
> it would default to an undefined, so I keep the same

I'm confuse, current code default to `''`

Reply:
very sorry, I forgot to push:

- Before: if no env variable is set, it is `undefined`;
- After: if `safeParse` fails, it falls back on `undefined`

Reply:
Thx